src/layout/constraint.rs

```diff
@@ -203,6 +332,23 @@ impl Constraint {
     {
         mins.into_iter().map(Constraint::Min).collect_vec()
     }
+
+    /// Convert an iterator of fill scaling factors into a vector of constraints
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use ratatui::prelude::*;
+    /// # let area = Rect::default();
+    /// let constraints = Constraint::from_mins([1, 2, 3]);
+    /// let layout = Layout::default().constraints(constraints).split(area);
+    /// ```
+    pub fn from_fills<T>(fills: T) -> Vec<Constraint>
```

Comment:
I think this makes sense calling it from relative lengths.

Reply:
I changed the name to `Proportional` now and updated these functions.